src/dicomweb_client/uri.py

```diff
@@ -629,12 +629,13 @@ def _validate_resource_identifiers_and_suffix(
 
 def _validate_uid(uid: str, permissive: bool) -> None:
     """Validates a DICOM UID."""
-    if len(uid) > _MAX_UID_LENGTH:
-        raise ValueError('UID cannot have more than 64 chars. '
-                         f'Actual count in {uid!r}: {len(uid)}')
-    if not permissive and _REGEX_UID.fullmatch(uid) is None:
-        raise ValueError(f'UID {uid!r} must match regex {_REGEX_UID!r} in '
-                         'conformance with the DICOM Standard.')
+    if not permissive:
```

Comment:
This was removed from the old `_assert_uid_format`, probably better to keep this check there

```suggestion
    if not isinstance(uid, str):
            raise TypeError('DICOM UID must be a string.')
    if not permissive:
```

Reply:
Done 👍🏽